tests/pykafka/test_protocol.py

```diff
@@ -256,5 +256,95 @@ def test_offset_fetch_response(self):
         self.assertEqual(response.topics[b'emmett.dummy'][0].offset, 1)
 
 
+class TestGroupMembershipAPI(unittest2.TestCase):
+    maxDiff = None
+
+    def test_consumer_group_protocol_metadata(self):
+        meta = protocol.ConsumerGroupProtocolMetadata()
+        msg = meta.get_bytes()
+        self.assertEqual(
+            msg,
+            bytearray(b'\x00\x00\x00\x00\x00\x01\x00\ndummytopic\x00\x00\x00\x0ctestuserdata')
```

Comment:
Sounds like a great idea. Again, the tests in this pull request are consistent with the rest of the existing protocol tests. I think overhauling the way we do protocol testing is outside the scope of this pull request. I've opened an [issue](https://github.com/Parsely/pykafka/issues/475) about this as well.